lib/Telemetry/src/DataQueue.cpp

```diff
@@ -3,29 +3,35 @@
 
 #include "settings.h"
 
-DataQueue::DataQueue() {
+DataQueue::DataQueue(String publishHeader) {
+	_publishHeader = publishHeader;
 	_init();
 }
 
 void DataQueue::add(String id, int value) {
-	_writer->beginObject()
-    	.name("t").value(id)
-    	.name("d").value(value)
-    .endObject();
+	_writer->name(id).value(value);
 }
 
 void DataQueue::add(String id, String value) {
-	_writer->beginObject()
-    	.name("t").value(id)
-    	.name("d").value(value)
-    .endObject();
+	_writer->name(id).value(value);
 }
 
 void DataQueue::add(String id, float value) {
+	_writer->name(id).value(value);
+}
+
+void DataQueue::add(String id, double value) {
+	_writer->name(id).value(String(value));
+}
```

Comment:
That makes a lot of sense. I think I wrote the original code for this when I didn't know about templates 😆

Reply:
I have mixed feelings about this because we're restricted to logging the types which JsonWriter::value is able to consume.  I ran into a situation at EA not too long ago where two engineers were debating the merits of strictly typed telemetry logging functions vs dynamically typed ones.  In the end we did go for dynamic typing, but the main point against it was that it would be more prone to breakage at runtime.

I guess it doesn't really matter doing something like that in a program as static as ours because we'd hopefully see those breakages when we compile.  Maybe this is a larger debate though

Reply:
I don't think we'll run into that issue here since the types are function calls are resolved at compile time. If the templated type doesn't resolve properly for any of the code within the function, the compiler should complain. For example:

```cpp
#include <iostream>

void is(int _) {
  std::cout << "is int" << std::endl;
}

void is(bool _) {
  std::cout << "is bool" << std::endl;
}

template <class T>
void add(T t) {
  is(t);
}

int main() {
  add<bool>(true); // prints is bool
  add<int>(5); // prints is int
  add<std::string>("string"); // compile error
} 
```